webapp/application.py

```diff
@@ -125,4 +131,178 @@ def application_page(token):
         application=application,
         job=job,
         hiring_lead=hiring_lead,
+        token=token,
+    )
+
+
+@application.route("/withdraw/<string:token>")
+def application_withdrawal(token):
+    payload = verification_token_cipher.decrypt(token)
+    try:
+        payload = json.loads(payload)
+        email = payload.get("email")
+        withdrawal_reason = payload.get("withdrawal_reason", "")
+        candidate_id = payload.get("candidate_id")
+        application_id = payload.get("application_id")
+    except (ValueError, TypeError):
+        flask.abort(404)
+
+    if not (email and candidate_id and application_id):
+        flask.abort(404)
+
+    application = harvest.get_application(application_id)
+    candidate = harvest.get_candidate(application["candidate_id"])
+    if (
+        "candidate_id" not in application
+        or application["candidate_id"] != candidate_id
+    ):
+        flask.abort(404)
+
+    print(candidate["first_name"], withdrawal_reason)
+
+    # call the Harvest API to reject the application
+
+    rejection_reason_id = get_reason_id(withdrawal_reason)
+    notes = withdrawal_reason
+
+    harvest.reject_application(
+        application_id, candidate_id, rejection_reason_id, notes
+    )
+
+    print(application_id, candidate_id, rejection_reason_id, notes)
```

Comment:
If it's important that we track these rejections, could we submit them to Sentry rather than the logs? In the logs they'll get immediately buried, and will be deleted after about 4 days.